src/modules/ekf2/ekf2_main.cpp

```diff
@@ -489,11 +491,45 @@ void Ekf2::task_main()
 			ctrl_state.pitch_rate = _lp_pitch_rate.apply(sensors.gyro_rad_s[1]);
 			ctrl_state.yaw_rate = _lp_yaw_rate.apply(sensors.gyro_rad_s[2]);
 
+			// Velocity in body frame
+			float velocity[3];
+			_ekf->copy_velocity(velocity);
+			Vector3f v_n(velocity);
+			matrix::Dcm<float> R_to_body(q.inversed());
+			Vector3f v_b = R_to_body * v_n;
+			ctrl_state.x_vel = v_b(0);
+			ctrl_state.y_vel = v_b(1);
+			ctrl_state.z_vel = v_b(2);
+
+
+			// Local Position NED
+			float position[3];
+			_ekf->copy_position(position);
+			ctrl_state.x_pos = position[0];
+			ctrl_state.y_pos = position[1];
+			ctrl_state.z_pos = position[2];
+
+			// Attitude quaternion
 			ctrl_state.q[0] = q(0);
 			ctrl_state.q[1] = q(1);
 			ctrl_state.q[2] = q(2);
 			ctrl_state.q[3] = q(3);
 
+			// Acceleration data
+			matrix::Vector<float, 3> acceleration = {&sensors.accelerometer_integral_m_s[0]};
+			acceleration = acceleration / sensors.accelerometer_integral_dt[0];
```

Comment:
I think for the control state using the low passed accel data makes more sense. @tumbili Can you rebase change that? Thanks!